cmd/mesh-bootstrap/main.go

```diff
@@ -0,0 +1,158 @@
+	p2pnet "github.com/libp2p/go-libp2p-net"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/plaid/go-envvar/envvar"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// peerCountLow is the target number of peers to connect to at any given time.
+	peerCountLow = 1000
+	// peerCountHigh is the maximum number of peers to be connected to. If the
+	// number of connections exceeds this number, we will prune connections until
+	// we reach peerCountLow.
+	peerCountHigh = 1100
+	// peerGraceDuration is the amount of time a newly opened connection is given
+	// before it becomes subject to pruning.
+	peerGraceDuration = 10 * time.Second
+	// defaultNetworkTimeout is the default timeout for network requests (e.g.
+	// connecting to a new peer).
+	defaultNetworkTimeout = 30 * time.Second
+)
+
+// Config contains configuration options for a Node.
+type Config struct {
+	// Verbosity is the logging verbosity: 0=panic, 1=fatal, 2=error, 3=warn, 4=info, 5=debug 6=trace
+	Verbosity int `envvar:"VERBOSITY" default:"6"`
+	// P2PListenPort is the port on which to listen for new connections. It can be
+	// set to 0 to make the OS automatically choose any available port.
+	P2PListenPort int `envvar:"P2P_LISTEN_PORT" default:"0"`
+	// PrivateKey path is the path to a private key file which will be used for
+	// signing messages and generating a peer ID.
+	PrivateKeyPath string `envvar:"PRIVATE_KEY_PATH" default:"0x_mesh/keys/privkey"`
+}
+
+func main() {
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Parse env vars
+	var config Config
+	if err := envvar.Parse(&config); err != nil {
+		panic(fmt.Sprintf("could not parse environment variables: %s", err.Error()))
+	}
+
+	// Configure logger to output JSON
+	// TODO(albrow): Don't use global settings for logger.
+	log.SetFormatter(&log.JSONFormatter{})
+	log.SetLevel(log.Level(config.Verbosity))
+
+	// Parse private key file
+	privKey, err := initPrivateKey(config.PrivateKeyPath)
+	if err != nil {
+		log.WithField("error", err).Fatal("could not initialize private key")
+	}
+
+	// Set up the transport and the host.
+	// Note: 0.0.0.0 will use all available addresses.
+	hostAddr, err := ma.NewMultiaddr(fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", config.P2PListenPort))
+	if err != nil {
+		log.WithField("error", err).Fatal("could not parse multiaddr")
+	}
+	connManager := connmgr.NewConnManager(peerCountLow, peerCountHigh, peerGraceDuration)
+	opts := []libp2p.Option{
+		libp2p.ListenAddrs(hostAddr),
+		libp2p.Identity(privKey),
+		libp2p.ConnectionManager(connManager),
+	}
+	basicHost, err := libp2p.New(ctx, opts...)
+	if err != nil {
+		log.WithField("error", err).Fatal("could not create host")
+	}
+
+	// Set up the notifee.
+	basicHost.Network().Notify(&notifee{})
+
+	// Set up DHT for peer discovery.
+	kadDHT, err := dht.New(ctx, basicHost)
+	if err != nil {
+		log.WithField("error", err).Fatal("could not create DHT")
+	}
+
+	if err := kadDHT.Bootstrap(ctx); err != nil {
```

Comment:
Since we have no nodes to bootstrap against, the routing tables of these peers will remain empty until the first peers connect to them -- which I guess it's the behaviour you're looking for. Just wanted to point it out.